integration/build_dependencies_test.go

```diff
@@ -17,14 +17,97 @@ limitations under the License.
 package integration
 
 import (
+	"fmt"
 	"strings"
 	"testing"
 
+	"github.com/google/uuid"
 	"github.com/sirupsen/logrus"
 
 	"github.com/GoogleContainerTools/skaffold/integration/skaffold"
 )
 
+func TestBuild_WithCache(t *testing.T) {
+	MarkIntegrationTest(t, CanRunWithoutGcp)
+
+	tests := []struct {
+		description string
+		rebuilt     []int
+		cached      []int
+		change      []int
+	}{
+		{
+			description: "no change",
+			cached:      []int{1, 2, 3, 4},
+		},
+		{
+			description: "change 1",
+			rebuilt:     []int{1},
+			change:      []int{1},
+			cached:      []int{2, 3, 4},
+		},
+		{
+			description: "change 2",
+			change:      []int{2},
+			rebuilt:     []int{1, 2},
+			cached:      []int{3, 4},
+		},
+		{
+			description: "change 3",
+			change:      []int{3},
+			rebuilt:     []int{1, 2, 3},
+			cached:      []int{4},
+		},
+		{
+			description: "change 4",
+			change:      []int{4},
+			rebuilt:     []int{4},
+			cached:      []int{1, 2, 3},
+		},
+		{
+			description: "change all",
+			change:      []int{1, 2, 3, 4},
+			rebuilt:     []int{1, 2, 3, 4},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			skaffold.Build("--cache-artifacts=true").InDir("testdata/build-dependencies").RunOrFail(t)
+			checkImagesExist(t)
+
+			// modify file `foo` to invalidate cache for target artifacts
+			for _, i := range test.change {
+				Run(t, fmt.Sprintf("testdata/build-dependencies/app%d", i), "sh", "-c", fmt.Sprintf("echo %s > foo", uuid.New().String()))
+			}
+			out, err := skaffold.Build("--cache-artifacts=true").InDir("testdata/build-dependencies").RunWithCombinedOutput(t)
+
+			if err != nil {
```

Comment:
Drop the missing line?
```suggestion
			out, err := skaffold.Build("--cache-artifacts=true").InDir("testdata/build-dependencies").RunWithCombinedOutput(t)
			if err != nil {
```